### PR TITLE
kubectl_build: build docker images directly in the k8s cluster

### PR DESCRIPTION
**Pull request by glehmann**

### CODE REVIEW

**Comment by glehmann** [comment]
@victorwuky please review :)

**Comment by nicks** [comment]
ooh very excited for this! will try to review later this afternoon

**Review by nicks — APPROVED** [review]
LGTM!

File: kubectl_build/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* `dockerfile_contents`
+* `only`
+* `container_args`
+* `network`
+
+some of these arguments may be added in the future (a pull request is welcome!).
+
+BuildKit CLI for kubectl doesn't support `env` secrets at this time, so you have to convert them to file to use this extension.
+
+BuildKit CLI for kubectl doesn't support all the local k8s clusters — k3d, for example, is know to not be supported.
```
[the remark this answers is not in the document]
Comment (glehmann) [reply kubectl_build/README.md:49]: The problem is more with how k3d mounts the filesystem, according to https://github.com/vmware-tanzu/buildkit-cli-for-kubectl/blob/main/docs/installing.md#k3d
I had to swith from k3d to kind to be able to use Buildkit CLI for kubectl, but kind's image a lot heavier than k3d's one, so I'd be glab if we can make it work. But given the description of the problem in that link, I'm not very optimitic!

Thanks for accepting the PR :)

**Pull request merged**